src/vs/workbench/services/preferences/common/preferencesUrlHandler.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URI } from 'vs/base/common/uri';
+import { IURLHandler } from 'vs/platform/url/common/url';
+import { IOpenSettingsOptions, IPreferencesService } from 'vs/workbench/services/preferences/common/preferences';
+
+export const SettingsAuthority = 'settings';
+
+export class PreferencesUrlHandler implements IURLHandler {
+	constructor(
+		@IPreferencesService private readonly preferencesService: IPreferencesService
+	) { }
+
+	/**
+	 * Should be of the format:
+	 * 	vscode://settings/settingName/optionalSettingValue
+	 * Examples:
+	 * 	vscode://settings/files.autoSave/afterDelay
```

Comment:
vscode or just code? I notice that the example above uses code-oss, and that the regex accepts both. Is there any filtering later to ensure that only values such as code, code-insiders, code-oss, and code-exploration are accepted?

Reply:
That is defined by `product.json` property `urlProtocol` and is dependent on the quality.

Reply:
☝️what @bpasero said. I just used `vscode` in the comment. I made the regex pretty open, and it should accept anything that we build: `code-oss`, `code`, `code-insiders`, and whatever exploration does. I'll change the comment to use `code`

Reply:
I've updated the regex to be more specific and use the product service for the protocol.

Reply:
Actually, using the exact protocol here isn't nice since we'll end up not doing the special settings rendering in vscode-insiders that way. I will keep the `.*code.*` regex.